source/MRMesh/MRCone3.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include "MRLine.h"
+
+namespace MR
+{
+
+// Base class for cone parameterization
+
+template <typename T>
+class Cone3
+{
+public:
+    Cone3()
+    {}
+
+    Cone3( const Line3<T>& inAxis, T inAngle, T inLength )
+        :
+        position( inAxis ),
+        angle( inAngle ),
+        length( inLength )
+    {}
+
+    inline MR::Vector3<T>& center( void )
```

Comment:
Need to add a second `const` overload. Or, if you don't use `Line3` and store two points separately, you can remove this function altogether.

Also, `inline` has no effect here (when a function is defined directly in a class body). I'd drop `void` too...

Reply:
Usage Line was a Fedor request. 
We could possibly use Line3  

Thanks for comment